src/worker/move-contact-worker.ts

```diff
@@ -0,0 +1,160 @@
+      const response = await axios.get(`${instanceUrl}/api/v2/monitoring`);
+      const sentinelBacklog = response.data.sentinel?.backlog;
+      return sentinelBacklog < this.MAX_SENTINEL_BACKLOG;
+    } catch (err: any) {
+      const errorMessage = err.response?.data?.error?.message || err.response?.data || err?.message;
+      console.error('Error fetching monitoring data:', errorMessage);
+      return false;
+    }
+  }
+
+  private async moveContact(jobData: MoveContactData): Promise<JobResult> {
+    try {
+      const { contactId, parentId, instanceUrl, sessionToken } = jobData;
+
+      if (!sessionToken) {
+        return { success: false, message: 'Missing session token' };
+      } else if (!queuePrivateKey) {
+        return { success: false, message: 'Missing QUEUE_PRIVATE_KEY' };
+      }
+
+      const decodedToken = Auth.decodeToken(sessionToken, queuePrivateKey);
+      const token = decodedToken.sessionToken.replace('AuthSession=', '');
+
+      const command = 'cht';
+      const args = this.buildCommandArgs(instanceUrl, token, contactId, parentId);
+      
+      this.logCommand(command, args);
+      await this.executeCommand(command, args);
+
+      return { success: true, message: `Job processing completed.` };
+    } catch (error) {
+      return { success: false, message: error as string };
+    }
+  }
+
+  private buildCommandArgs(instanceUrl: string, sessionToken: string, contactId: string, parentId: string): string[] {
+    return [
+      `--url=${instanceUrl}`,
+      `--session-token=${sessionToken}`,
+      '--force',
+      'move-contacts',
+      'upload-docs',
+      '--',
+      `--contacts=${contactId}`,
+      `--parent=${parentId}`
+    ];
+  }
+
+  private logCommand(command: string, args: string[]): void {
+    const maskedArgs = args.map(arg => arg.startsWith('--session-token=') ? '--session-token=********' : arg);
+    console.log('Executing command:', `${command} ${maskedArgs.join(' ')}`);
+  }
+
+  private async executeCommand(command: string, args: string[]): Promise<void> {
+    return new Promise((resolve, reject) => {
+      const chtProcess = spawn(command, args);
+
+      const timeout = setTimeout(() => {
+        chtProcess.kill();
+        reject(new Error('cht-conf timed out'));
+      }, this.MAX_TIMEOUT);
+
+      chtProcess.stdout.on('data', data => {
+        console.log(`cht-conf: ${data}`);
+      });
+
+      chtProcess.stderr.on('data', error => {
+        console.error(`cht-conf error: ${error}`);
+      });
+
+      chtProcess.on('close', code => {
+        clearTimeout(timeout);
+        code === 0 ? resolve() : reject(new Error(`cht-conf exited with code ${code}`));
+      });
+
+      chtProcess.on('error', error => {
+        clearTimeout(timeout);
+        reject(error);
+      });
+    });
+  }
+
+  private async postpone(job: Job): Promise<boolean> {
+    // Add this job back to queue with a DELAY_IN_MILLIS delay
+    const jobParams = {
```

Comment:
i tested this today and i'm unsure if it is working.  i had about 50 move-contact commands and they ran until sentinel backlog was >1000.  i waited well over an hour but they did not run again.